### PR TITLE
Expanding StringTokenizator with ReadOnlySpan.

### PR DESCRIPTION
**Pull request by Meloman19**
## What does the pull request do?
Returns `ReadOnlySpan<char>` as `CurrentToken` instead of string substring.

## What is the current behavior?
`CurrentToken` always create new string instance (via `Substring`).

## What is the updated/expected behavior with this PR?
In new runtimes (greater than Standard), this reduces allocations by parsing `int` and `double` directly from Span.

### CODE REVIEW

File: src/Avalonia.Base/Utilities/StringTokenizer.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public StringTokenizer(string s, char separator = DefaultSeparatorChar, string?
             }
         }
 
-        public string? CurrentToken => _tokenIndex < 0 ? null : _s.Substring(_tokenIndex, _tokenLength);
+        public ReadOnlySpan<char> CurrentToken => _tokenIndex < 0 ? ReadOnlySpan<char>.Empty : _s.AsSpan().Slice(_tokenIndex, _tokenLength);
```
Comment (maxkatz6) [inline src/Avalonia.Base/Utilities/StringTokenizer.cs:47]: StringTokenizer is part of public API, for no good reason.
Which unfortunately is the reason why we can't change this property type.
As an alternative, you should be able to add `ReadOnlySpan<char> CurrentTokenSpan` property and use it instead.
While keeping old property as a string.

Comment (maxkatz6) [reply src/Avalonia.Base/Utilities/StringTokenizer.cs:47]: This class should be removed from public API in 12.0. But we are not there yet.

Comment (maxkatz6) [reply src/Avalonia.Base/Utilities/StringTokenizer.cs:47]:
```
##[error]API compatibility errors between 'baseline/netstandard2.0/Avalonia.Base.dll' (left) and 'target/netstandard2.0/Avalonia.Base.dll' (right):
##[error]CP0002: Member 'string? Avalonia.Utilities.StringTokenizer.CurrentToken.get' exists on baseline/netstandard2.0/Avalonia.Base.dll but not on target/netstandard2.0/Avalonia.Base.dll
##[error]API breaking changes found. If those are intentional, the APICompat suppression file can be updated by specifying the '--generate-suppression-file' parameter.
##[error]Target "ValidateApiDiff" has thrown an exception
```

Comment (Meloman19) [reply src/Avalonia.Base/Utilities/StringTokenizer.cs]: @maxkatz6 done. 
Changing `[MaybeNullWhen(false)]` in `TryReadString` also affects the API? Or doesn't matter what value is returned from the "false" `Try*` function?

Comment (maxkatz6) [reply src/Avalonia.Base/Utilities/StringTokenizer.cs]: Changing attributes is not a breaking change.

**Review by maxkatz6 — APPROVED** [review]
Thanks!

**Pull request merged**